composer.json

```diff
@@ -25,14 +25,34 @@
         "ockcyp/covers-validator": "1.3.3",
         "phpunit/phpunit": "^8.5.13||^9.5.0",
         "php-parallel-lint/php-console-highlighter": "0.5",
-        "php-parallel-lint/php-parallel-lint": "1.3.1"
+        "php-parallel-lint/php-parallel-lint": "1.2.0",
+        "friendsofphp/php-cs-fixer": "^3.5",
+        "brainmaestro/composer-git-hooks": "^2.8"
     },
     "scripts": {
+        "post-install-cmd": "cghooks add --ignore-lock",
+        "post-update-cmd": "cghooks update",
         "test": [
             "parallel-lint . --exclude vendor",
             "covers-validator",
             "phpunit --coverage-text"
         ],
-        "cover": "phpunit --coverage-html coverage"
+        "cover": "phpunit --coverage-html coverage",
+        "cghooks": "vendor/bin/cghooks",
+        "fix": "php-cs-fixer fix --config=.php-cs-fixer.php"
+    },
+    "extra": {
+        "hooks": {
+            "config": {
+                "stop-on-failure": ["pre-commit"]
+            },
+            "pre-commit": [
+                "composer run-script fix",
+                "git add -A"
```

Comment:
I'd be wary of using `git add -A`; but I don't know offhand if there's a better option.

For example, I might have a file laying around in the repo (that I didn't want to add/commit), or changes to a file that I didn't want to commit either, running `git add -A` (or similarly `git commit -a`) would result in these being included hwen it might not be wanted